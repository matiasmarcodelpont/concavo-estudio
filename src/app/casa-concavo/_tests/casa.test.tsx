import '@testing-library/jest-dom'
import { render, screen, within } from '@testing-library/react'

import CasaConcavo from '@/app/casa-concavo/page'

jest.mock('@/data/data.ts', () => ({
  ambientes: [
    {
      slug: 'cocina',
      name: 'Cocina',
    },
    {
      slug: 'living-comedor',
      name: 'Living/Comedor',
    },
  ],
  colaboradores: [
    {
      slug: 'elegance',
      name: 'Elegance',
      website: 'www.pisoselegance.es',
      isMain: true,
      description:
        'Líder en revestimientos de suelos elegantes y duraderos para hogares y negocios. Calidad, innovación y compromiso con la sostenibilidad nos distinguen.',
      email: 'info@pisoselegance.com',
      address: 'Avenida del Diseño 123, Ciudad Elegante, España',
    },
    {
      slug: 'luz-viva',
      name: 'Luz Viva',
      website: 'www.luzviva.com',
      isMain: true,
      description:
        'Líder en sistemas de iluminación innovadores y eficientes para hogares y negocios. Calidad, diseño y compromiso con la sostenibilidad nos distinguen.',
      email: 'info@luzviva.com',
      address: 'Calle de la Luz 456, Ciudad Brillante, México',
    },
    {
      slug: 'muebles-arte',
      name: 'Muebles de Arte',
      website: 'www.mueblesdearte.com',
      isMain: false,
      description: null,
      email: null,
      address: 'Calle del Diseño 789, Ciudad Creativa, Argentina',
    },
    {
      slug: 'cocina-design',
      name: 'Cocina Design',
      website: 'www.cocinadesign.com',
      isMain: false,
      description:
        'Especialistas en diseño y equipamiento de cocinas de alta gama. Ofrecemos soluciones personalizadas para crear la cocina de tus sueños, combinando estilo, funcionalidad y calidad.',
      email: null,
      address: null,
    },
  ],
}))

describe('Casa Cóncavo', () => {
  it('renders Casa Cóncavo unchanged', () => {
    const { container } = render(<CasaConcavo />)
    expect(container).toMatchSnapshot()
  })

  it('renders Casa Cóncavo ambientes', () => {
    render(<CasaConcavo />)

    const ambientesList = screen.getByRole('list', { name: 'Lista de Ambientes' })
    expect(ambientesList).toBeInTheDocument()

    const ambientes = within(ambientesList).getAllByRole('listitem')
    expect(ambientes).toHaveLength(2)

    expect(ambientes[0]).toBeInTheDocument()
    const link0 = within(ambientes[0]).getByRole<HTMLAnchorElement>('link', { name: 'Cocina' })
    expect(link0).toBeInTheDocument()
    expect(link0.href).toMatch(/^https?:\/\/[^/]+\/casa\/cocina$/)

    expect(ambientes[1]).toBeInTheDocument()
    const link1 = within(ambientes[1]).getByRole<HTMLAnchorElement>('link', { name: 'Living/Comedor' })
    expect(link1).toBeInTheDocument()
    expect(link1.href).toMatch(/^https?:\/\/[^/]+\/casa\/living-comedor$/)
  })
})
