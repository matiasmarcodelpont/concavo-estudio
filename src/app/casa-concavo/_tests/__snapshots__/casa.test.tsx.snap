// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Casa Cóncavo renders Casa Cóncavo unchanged 1`] = `
<div>
  <main>
    <section
      class="pb-24 sm:pb-32 md:pb-40 pt-28 sm:pt-36 md:pt-48 flex"
    >
      <div
        class="flex-grow-[1]"
      />
      <article
        class="relative bg-white text-sm sm:text-base md:text-lg text-left p-[1.75em] mx-8 w-fit max-w-[330px] sm:max-w-[380px] md:max-w-[420px]"
      >
        <h1
          class="font-extrabold inline"
        >
          CASA CÓNCAVO
        </h1>
         es nuestra forma de vivir las casa pensando en el todo y en el detalle. Conocé nuestra primera casa: ´Whitehaven´.
        <div
          class="absolute -top-3 md:-top-4 -left-0"
        >
          <div
            class="absolute w-[0.5px] h-6 md:h-8 bg-black"
          />
          <div
            class="absolute w-[0.5px] h-6 md:h-8 bg-black rotate-90"
          />
        </div>
      </article>
      <div
        class="flex-grow-[2]"
      />
    </section>
    <section>
      <ul
        aria-label="Lista de Ambientes"
        class="space-y-2"
      >
        <li>
          <a
            aria-label="Cocina"
            class="relative block w-full h-[40vw] min-h-[300px]"
            href="/casa-concavo/cocina"
          >
            <img
              alt="Cocina"
              class="-z-10"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=3840&q=75"
              srcset="/_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=640&q=75 640w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=750&q=75 750w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=828&q=75 828w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
            <div
              class="w-full h-full bg-[#0003] p-10"
            >
              <span
                style="font-size: 0px;"
              />
              <h2
                class="font-editorial-new tracking-[0.01em] uppercase mt-12 mb-6 sm:mt-16 sm:mb-9 md:mt-20 md:mb-12 opacity-0 text-2xl sm:text-3xl md:text-4xl text-bone text-left font-extralight"
              >
                Cocina
              </h2>
            </div>
            <span
              style="font-size: 0px;"
            />
          </a>
        </li>
        <li>
          <a
            aria-label="Living/Comedor"
            class="relative block w-full h-[40vw] min-h-[300px]"
            href="/casa-concavo/living-comedor"
          >
            <img
              alt="Living/Comedor"
              class="-z-10"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=3840&q=75"
              srcset="/_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=640&q=75 640w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=750&q=75 750w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=828&q=75 828w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
            <div
              class="w-full h-full bg-[#0003] p-10"
            >
              <span
                style="font-size: 0px;"
              />
              <h2
                class="font-editorial-new tracking-[0.01em] uppercase mt-12 mb-6 sm:mt-16 sm:mb-9 md:mt-20 md:mb-12 opacity-0 text-2xl sm:text-3xl md:text-4xl text-bone text-left font-extralight"
              >
                Living/Comedor
              </h2>
            </div>
            <span
              style="font-size: 0px;"
            />
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
